### PR TITLE
4.x : Workspace Refactoring : load multiple workspaces

### PR DESCRIPTION
**Pull request by deepakrkris**
Modify loopback models to handle multiple workspaces

### CODE REVIEW

File: component/datamodel/util/file-utility.js [hunk 1]
```diff
@@ -5,10 +5,13 @@
 'use strict';
 
 const async = require('async');
+const config = require('../../config');
 const fs = require('fs-extra');
+const glob = require('glob');
 const path = require('path');
 
 module.exports = {
+  listOfFiles: listOfFiles,
```
Comment (superkhau) [inline component/datamodel/util/file-utility.js]: listFiles/getFileNames?

File: component/datamodel/util/file-utility.js [hunk 2]
```diff
@@ -121,3 +124,47 @@ function readMiddleware(workspace, cb) {
...
+        // flatten paths into single list
+        let merged = [];
+        merged = merged.concat.apply(merged, paths);
+        result[key] = merged;
+        next();
+      });
+    });
+  });
+
+  async.parallel(steps, function(err, data) {
```
Comment (superkhau) [inline component/datamodel/util/file-utility.js]: remove unused var `, data`

File: component/tasks.js [hunk 3]
```diff
@@ -112,6 +114,20 @@ class Tasks {
...
     fsUtility.writeDataSourceConfig(workspace, cb);
   }
+  loadModel(filePath, fileData, next) {
+    const workspace = this;
+    const fileName = path.basename(filePath);
+    const modelName = lodash.capitalize(lodash.camelCase(fileName));
+    const facetName = fileData.facetName;
+    if (facetName && modelName) {
+      workspace.refreshModel(filePath, function(err) {
+        if (err) return next(null, err);
```
Comment (superkhau) [inline component/tasks.js]: why is err the second arg?

File: component/datamodel/util/file-utility.js (hunk 2 above)
Comment (deepakrkris) [inline component/datamodel/util/file-utility.js]: agreed

**Pull request merged**